src/mockgyver.erl

```diff
@@ -1372,73 +1372,103 @@ ensure_renamed_mod_to_load_aux(RenamedMod, #modinfo{code=Code}) ->
-                                none,
-                                [mk_call('ActionFun', Args)])])],
-    erl_syntax:function(
-      erl_syntax:abstract(F),
-      [erl_syntax:clause(Args, none, Body)]).
+    %% It is many times faster than constructing a number of functions,
+    %% each containing the inner case expression, though.
+    func_from_str_fmt(
+      "'$handle_undefined_function'(FnName, Args) ->
+           Arity = length(Args),
+           case lists:member({FnName, Arity}, ~p) of
+               true ->
+                   case mockgyver:reg_call_and_get_action({~p,FnName,Args}) of
+                       undefined ->
+                           ~s;
+                       ActionFun ->
+                          try
+                              apply(ActionFun, Args)
+                          catch
+                              Class:Reason:St0 ->
+                                  ToMF = {~p, FnName},
+                                  St = rewrite_st(ActionFun, ToMF, Arity, St0),
+                                  erlang:raise(Class, Reason, St)
+                          end
+                   end;
+               false ->
+                   error_handler:raise_undef_exception(~p, FnName, Args)
+           end.",
+      [ExportedFAs, Mod, FmtNoAction("FnName", "Args"), Mod, Mod]).
+
+%% Ensure a stacktrace like the following contains the mocked module
+%% and function, instead of an internal fun that the user was not
+%% involved in creating.
+%%
+%% Test code:
+%%
+%%     ?WHEN(mockgyver_dummy:return_arg(N) -> error(foo)),
+%%     mockgyver_dummy:return_arg(1),
+%%
+%% Before:
+%%
+%%      Failure/Error: {error,function_clause,
+%%                         [{mockgyver_tests,
+%%                              '-some_test/1-fun-0-',
+%%                              [-1],
+%%                              [...]},
+%%                          ...
+%%
+%% After:
+%%
+%%      Failure/Error: {error,function_clause,
+%%                         [{mockgyver_dummy,
+%%                              return_arg,
+%%                              [-1],
+%%                              [...]},
+%%                          ...
+mk_rewrite_st_function(Mod) ->
+    func_from_str_fmt(
+      "rewrite_st(ActionFun, ToMF, Arity, Stacktrace)
+         when is_function(ActionFun) ->
+           {module, FromM} = erlang:fun_info(ActionFun, module),
+           {name, FromF} = erlang:fun_info(ActionFun, name),
+           rewrite_st({FromM, FromF}, ToMF, Arity, Stacktrace);
+       rewrite_st({FromM, FromF}, {ToM, ToF}, Arity, Stacktrace) ->
+           [case {{M, F}, A} of
+                {{FromM, FromF}, A} when A == Arity; length(A) == Arity ->
+                    {ToM, ToF, A, Extra};
+                _MFA ->
+                    StElem
+            end || {M, F, A, Extra}=StElem <- Stacktrace,
+            %% Exclude the error handler function from the stacktrace
+            {M, F} /= {~p, '$handle_undefined_function'}].",
```

Comment:
Nitpick: Indent a few spaces to indicate it is a filter expression? (I first thought this function clause returned a boolean :) )

Reply:
Oups. Fixed in next revision.